lib/libshare/smb.c

```diff
@@ -153,39 +160,45 @@ smb_retrieve_shares(void)
 					goto out;
 				}
 
-				strncpy(shares->name, name,
-					sizeof (shares->name));
-				shares->name [sizeof (shares->name) - 1] = '\0';
+				(void) strlcpy(shares->name, name,
+						sizeof (shares->name));
 
-				strncpy(shares->path, path,
-				    sizeof (shares->path));
-				shares->path [sizeof (shares->path) - 1] = '\0';
+				(void) strlcpy(shares->path, path,
+						sizeof (shares->path));
 
-				strncpy(shares->comment, comment,
-				    sizeof (shares->comment));
-				shares->comment[sizeof (shares->comment)-1] =
-				    '\0';
+				(void) strlcpy(shares->comment, comment,
+						sizeof (shares->comment));
```

Comment:
@behlendorf， @rlaager 
sorry，I didn't pay attention to this.
please review it again.
thank you!